src/main/java/com/es/phoneshop/model/product/ArrayListProductDao.java

```diff
@@ -1,68 +1,92 @@
 package com.es.phoneshop.model.product;
 
-import java.math.BigDecimal;
-import java.util.*;
+import com.es.phoneshop.enums.Order;
+import com.es.phoneshop.enums.SortBy;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 public class ArrayListProductDao implements ProductDao {
-
+    private static ArrayListProductDao instance;
     private Long counter = 0L;
-
     private List<Product> productList;
 
-    public ArrayListProductDao() {
-        this.productList = getSampleProducts();
-        productList.forEach(product -> product.setId(++counter));
+    private ArrayListProductDao() {
+        this.productList = new ArrayList<>();
+    }
+
+    public static synchronized ArrayListProductDao getInstance() {
+        if (instance == null) {
```

Comment:
Don't ever use "static synchronized" methods, I'll later explain why. And try to avoid synchronized keyword as much as possible. Please use one of singleton implementations suggested by Andrei or Me